environments/jvm/src/main/java/io/fission/Server.java

```diff
@@ -29,13 +40,33 @@
 
 	private static Logger logger = Logger.getGlobal();
 
-	@RequestMapping(value = "/", method = { RequestMethod.GET, RequestMethod.POST, RequestMethod.DELETE,
-			RequestMethod.PUT })
-	ResponseEntity<Object> home(RequestEntity<?> req) {
-		if (fn == null) {
-			return ResponseEntity.badRequest().body("Container not specialized");
-		} else {
-			return ((ResponseEntity<Object>) ((Function) fn).call(req, null));
+	@RequestMapping(value = "/", method = { RequestMethod.GET, RequestMethod.POST, RequestMethod.DELETE, RequestMethod.PUT })
+	ResponseEntity<Object> home(HttpServletRequest request) {
+
+		String body;
+		// Converting header in multimap
+		try {
+				MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+				Enumeration headerNames = request.getHeaderNames();
+				while (headerNames.hasMoreElements()) {
+					String key = (String) headerNames.nextElement();
+					String value = request.getHeader(key);
+					headers.put(key, singletonList(value));
+				 }
+
+				//Extract the body from the request
+				body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+				//Get the rawrequest
+				RequestEntity<String> rawRequest = new RequestEntity<String>(body,headers,HttpMethod.resolve(request.getMethod()), new URI(request.getRequestURI()));
```

Comment:
No, user can send any data that he wants. We are converting the received request in string format and then sending it to the function.

Reply:
I don't think it's a good way to solve the problem by converting the body to string type silently at the server. And from the [doc](https://docs.spring.io/spring/docs/4.2.5.RELEASE_to_4.3.0.RC1/Spring%20Framework%204.3.0.RC1/org/springframework/http/RequestEntity.html#getType--), it looks like the `getType()` will always return `string` instead of the original type.

Reply:
Also, load all request body into memory will increase the memory consumption and caused OOM.

Reply:
> spring was internally modifying the XML by trimming or removing details of the XML content.
Could you point out the doc point out such behavior?

Reply:
`return ((ResponseEntity<Object>) ((Function) fn).call(req, null));`

The original server just bypasses the req to the user function without any modification before the call. So what's the difference for doing this at user function and at server?

Reply:
Hey @anubhav6663 , can you answer @life1347 's question about compatibility? At first glance it seems like this change will just break existing Java functions that are using this environment.

Reply:
@life1347 We are using HttpServletRequest as replacement for RequestEntity, in the original implementation we were using RequestEntity which is provided by springframework and when we are passing this XML 
<Book>
	<name>Ashish</name>
	<subject>
		 <name>Math</name>
		 <name>Hindi</name>
	</subject>
</Book> 
It is returning {name=Ashish, subject={name=Hindi}} on req.getBody(), to prevent this from happening we are using lower level interface HttpServletRequest. By using this we are preventing the data loss. 
I think this is the limitation of springframework's RequestEntity. If we don't want to change the interface we have to raise a issue on springframework and wait for it to get resolved.